examples/opt_serving/generator.py

```diff
@@ -137,6 +137,50 @@ def load_model(self):
             self.num_gpus = alpa.get_global_cluster().num_devices
         logger.info(f"Loading model time: {load_time:.2f}")
 
+    # FIXME(Hao): the current batch uses the topp, reponse_length of the first sentence.
+    def pad_batch(self, inputs, pad_value=None):
+        if pad_value is None:
+            pad_value = self.pad
+        new_inputs = inputs
+        src_lens = [len(input) for input in inputs]
+        max_len = max(src_lens)
+        bs = len(inputs)
+
+        for new_input in new_inputs:
+            ori_len = len(new_input)
+            if len(new_input) < max_len:
+                new_input.extend([pad_value for _ in range(max_len - ori_len)])
+        # pad to bs = MAX_BS
+        if bs < MAX_BS:
+            new_inputs.extend([[pad_value for _ in range(max_len)] for _ in range(MAX_BS - bs)])
+        return new_inputs
+    
+    def __call__(
+        self,
+        inputs,
+        cache_ids,
+        pasts=None,
+    ):
+        assert len(inputs) == 1, "enforcing batch size 1 for now to avoid messy logic with caching past key values"
+        inputs = self.pad_batch(inputs)
+        tokens = [torch.LongTensor(t) for t in inputs]
+        lengths = [len(t) for t in inputs]
+        batches = self.get_batch_iterator(
+            dataset=self.build_dataset_for_inference(tokens, lengths),
+            max_tokens=None,
+            max_sentences=MAX_BS,
+            max_positions=None,
+            ignore_invalid_inputs=False,
+        ).next_epoch_itr(shuffle=False)
+        assert len(batches) == 1, "we have enforce inner bs = 1."
+        outputs = []
+        for idx, (batch, cache_id) in enumerate(zip(batches, cache_ids)):
+            input_ids = move_to_cuda(batch['src_tokens'])[:, :-1]
```

Comment:
We change the default torch device to CPU (while alpa still runs on GPU) to avoid the memory conflicts between torch and alpa.

Please update accordingly
https://github.com/alpa-projects/alpa/blob/6085a0877a999d60cfc4409e25ef51cd96703eea/examples/opt_serving/generator.py#L171-L172

Reply:
Updated!